lib/driver.js

```diff
@@ -287,15 +291,18 @@ class AndroidUiautomator2Driver extends BaseDriver {
 
   async startUiAutomator2Session () {
     // get device udid for this session
-    let {udid, emPort} = await helpers.getDeviceInfoFromCaps(this.opts);
+    const {udid, emPort} = await helpers.getDeviceInfoFromCaps(this.opts);
     this.opts.udid = udid;
     this.opts.emPort = emPort;
 
     // now that we know our java version and device info, we can create our
     // ADB instance
     this.adb = await androidHelpers.createADB(this.opts);
 
-    const apiLevel = await this.adb.getApiLevel();
+    // the initial access to adb may fail for the device being offline
+    const apiLevel = await retryInterval(ADB_INITIAL_RETRIES, 100, async () => {
+      return await this.adb.getApiLevel();
```

Comment:
return is redundant

Reply:
We want the API level to be returned from `retryInterval`, so we need to return it inside.

Reply:
I mean `await retryInterval(ADB_INITIAL_RETRIES, 100, async () => await this.adb.getApiLevel())` would be enough

Reply:
or even await retryInterval(ADB_INITIAL_RETRIES, 100, this.adb.getApiLevel.bind(this.adb)